sdk/tables/azure-data-tables/azure/data/tables/_table_service_client.py

```diff
@@ -0,0 +1,295 @@
+       :return: None
+       :rtype: None
+       :raises: ~azure.core.exceptions.HttpResponseError
+       """
+        props = TableServiceProperties(
+            logging=analytics_logging,
+            hour_metrics=hour_metrics,
+            minute_metrics=minute_metrics,
+            cors=cors
+        )
+        try:
+            return self._client.service.set_properties(props, **kwargs)  # type: ignore
+        except HttpResponseError as error:
+            process_table_error(error)
+
+    @distributed_trace
+    def create_table(
+            self,
+            table_name,  # type: str
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> TableClient
+        """Creates a new table under the current account.
+
+        :param table_name: The Table name.
+        :type table_name: str
+        :return: TableClient
+        :rtype: ~azure.data.tables.TableClient
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        _validate_table_name(table_name)
+
+        table_properties = TableProperties(table_name=table_name, **kwargs)
+        self._client.table.create(table_properties)
+        table = self.get_table_client(table_name=table_name)
+        return table
+
+    @distributed_trace
+    def delete_table(
+            self,
+            table_name,  # type: str
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        """Deletes the table under the current account
+
+        :param table_name: The Table name.
+        :type table_name: str
+        :return: None
+        :rtype: None
+        """
+        _validate_table_name(table_name)
+
+        self._client.table.delete(table=table_name, **kwargs)
+
+    @distributed_trace
+    def query_tables(
+            self,
+            filter,  # pylint: disable=W0622
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> ItemPaged[Table]
+        """Queries tables under the given account.
+        :param filter: Specify a filter to return certain tables
+        :type filter: str
+        :keyword int results_per_page: Number of tables per page in return ItemPaged
+        :keyword Union[str, list(str)] select: Specify desired properties of a table to return certain tables
+        :keyword dict parameters: Dictionary for formatting query with additional, user defined parameters
+        :return: A query of tables
+        :rtype: ItemPaged[Table]
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+        parameters = kwargs.pop('parameters', None)
+        if parameters:
+            selected = filter.split('@')[1]
+            for key, value in parameters.items():
+                if key == selected:
+                    filter = filter.split('@')[0].replace('@', value)  # pylint: disable = W0622
+
+        temp_select = kwargs.pop('select', None)
+        select = " "
+        if temp_select is not None and len(list(temp_select)) > 1:
+            select = ", ".join(temp_select)
```

Comment:
This might do strange things if a string is passed in, I think we can refactor to something like:
```python
user_select = kwargs.pop('select', None)

# Normally we don't like type checking, however it's very difficult to use
# Pythonic means to distinguish between a string and an iterable
if user_select and not isinstance(user_select, str):
    user_select = ", ".join(user_select)

query_options = QueryOptions(select=user_select)
```